File: zmscitizenview/src/api/models/Service.ts
import { Combinable } from "./Combinable";

/**
 *
 * @export
 * @interface Service
 */
export interface Service {
  /**
   *
   * @type {string}
   * @memberof Service
   */
  id: string;
  /**
   *
   * @type {string}
   * @memberof Service
   */
  name: string;
  /**
   *
   * @type {number}
   * @memberof Service
   */
  maxQuantity: number;
  /**
   * @type {Combinable}
   * @memberof Service
   */
  combinable?: Combinable;
  /**
   * @type {(string | number | null)}
   * @memberof Service
   */
  parentId: string | number | null;
  /**
   * @type {(number | null)}
   * @memberof Service
   */
  variantId: number | null;
  /**
   * @type {(boolean)}
   * @memberof Service
   */
  showOnStartPage?: boolean;
}

File: zmsautomation/src/test/java/zms/ataf/rest/dto/zmscitizenapi/Service.java
package zms.ataf.rest.dto.zmscitizenapi;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Data;

/**
 * Service model based on schema: zmsentities/schema/citizenapi/service.json
 */
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Service {

    private Integer id;
    private String name;
    private Integer maxQuantity;
    private Object combinable;

    @JsonProperty("public")
    private Boolean isPublic;

    @JsonProperty("parent_id")
    private Integer parentId;

    @JsonProperty("variant_id")
    private Integer variantId;
}
